### PR TITLE
(chore) update to use markdown-transform

### PR DESCRIPTION
**Pull request by dselman**
Signed-off-by: Dan Selman <danscode@selman.org>

### Changes
- Update to use `markdown-transform` for Slate <> Markdown conversion
- Removed markdown <> HTML conversion from plugins

### Flags
- Plugins are now purely for controlling Slate editor behaviour, they cannot extend markdown serialisation as that is handled by markdown-transform
- Do we want to remove the `List` plugin altogether and just bake that into the `SlateAsInputEditor`?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "url": "https://github.com/accordproject/markdown-editor.git"
   },
   "scripts": {
-    "test": "jest",
+    "test": "jest --passWithNoTests",
```
Comment (irmerk) [inline package.json]: Is this temporary until new tests are created?

File: src/plugins/list.js [hunk 2]
```diff
@@ -4,53 +4,7 @@ import React from 'react';
...
-      },
-    };
-
-    const newSchema = JSON.parse(JSON.stringify(schema));
-    newSchema.blocks = { ...schema.blocks, ...additions.blocks };
-    newSchema.document.nodes[0].match.push({ type: 'ol_list' });
-    newSchema.document.nodes[0].match.push({ type: 'ul_list' });
-    return newSchema;
-  });
+  const name = 'list';
```
Comment (irmerk) [inline src/plugins/list.js]: Is this still used? Not seeing it in this file.

**Comment by irmerk** [comment]
Curious why we remove the video plugin?

**Comment by irmerk** [comment]
I think moving the `List` plugin altogether and just baking that into the `SlateAsInputEditor` makes sense, it feels like it's an integral attribute that doesn't need to be abstracted into a plugin.

File: package.json (hunk 1 above)
Comment (dselman) [reply package.json]: Yes -- and it only seems to be needed for the Travis build. Not sure why.

File: src/plugins/list.js (hunk 2 above)
Comment (dselman) [reply src/plugins/list.js]: It's used by the `PluginManager` to deterministically sort the plugins so they get applied in the same order.

**Comment by dselman** [comment]
> Curious why we remove the video plugin?

Because the `markdown-transform` would need to be extended to handle the conversion from/to Slate.

**Pull request merged**